contracts/vesting/src/vesting/account/vesting_account.rs

```diff
@@ -0,0 +1,206 @@
+use crate::contract::NUM_VESTING_PERIODS;
+use crate::errors::ContractError;
+use crate::traits::VestingAccount;
+use config::defaults::DENOM;
+use cosmwasm_std::{Coin, Env, Order, Storage, Timestamp, Uint128};
+
+use super::Account;
+
+impl VestingAccount for Account {
+    fn locked_coins(
+        &self,
+        block_time: Option<Timestamp>,
+        env: &Env,
+        storage: &dyn Storage,
+    ) -> Result<Coin, ContractError> {
+        // Returns 0 in case of underflow.
+        Ok(Coin {
+            amount: Uint128::new(
+                self.get_vesting_coins(block_time, env)?
+                    .amount
+                    .u128()
+                    .saturating_sub(
+                        self.get_delegated_vesting(block_time, env, storage)?
```

Comment:
Aren't the coins we have delegated/bonded still technically locked though?

Reply:
Its a weird definition of `locked`, we might want to change it...

```
locked_coins returns the set of coins that are not spendable (can still be delegated tough) (i.e. locked), defined as the vesting coins that are not delegated or bonded.
```